include/rxmesh/kernels/attribute.cuh

```diff
@@ -94,6 +94,86 @@ __launch_bounds__(blockSize) __global__
     }
 }
 
+struct CustomMaxPair
+{
+    template <typename T>
+    __device__ __forceinline__ cub::KeyValuePair<int, T> operator()(
+        const cub::KeyValuePair<int, T>& a,
+        const cub::KeyValuePair<int, T>& b) const
+    {
+        return (b.value > a.value) ? b : a;
+    }
+};
+
+struct CustomMinPair
+{
+    template <typename T>
+    __device__ __forceinline__ cub::KeyValuePair<int, T> operator()(
+        const cub::KeyValuePair<int, T>& a,
+        const cub::KeyValuePair<int, T>& b) const
+    {
+        return (b.value < a.value) ? b : a;
+    }
+};
+template <class T, uint32_t blockSize, typename HandleT>
+__launch_bounds__(blockSize) __global__
+    void arg_max_kernel(const Attribute<T, HandleT> X,
+                    bool                        is_min,
+                    const uint32_t              num_patches,
+                    const uint32_t              num_attributes,
+                    cub::KeyValuePair<int, T>*  d_block_output,
+                    uint32_t                    attribute_id)
+{
+    using LocalT = typename HandleT::LocalT;
+
+    assert(X.get_num_attributes() == 1); //we can only take arg max for a scalar attribute
+
+    uint32_t p_id = blockIdx.x;
+    if (p_id < num_patches) {
+        const uint16_t element_per_patch = X.size(p_id);
+        cub::KeyValuePair<int, T> thread_val;
+        thread_val.value = std::numeric_limits<T>::lowest();
+        thread_val.key   = 0;
+        for (uint16_t i = threadIdx.x; i < element_per_patch; i += blockSize) {
+
+            if (X.get_patch_info(p_id).is_owned(LocalT(i)) &&
+                !X.get_patch_info(p_id).is_deleted(LocalT(i))) {
+
+                if (attribute_id != INVALID32 ) 
+                {
+                    cub::KeyValuePair<int, T> current_pair(i, X(p_id, i, attribute_id));
+                    if (is_min) 
```

Comment:
Rather than deciding the type of operation at run time, you can have it as an input to the function. So, `arg_max_kernel` will have additional template that specifies the operation i.e., 

```
template <..., typename Op>
void arg_max_kernel (.., Op op){
      thread_val = op(thread_val, current_pair);
}
```